front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields/index.tsx

```diff
@@ -59,31 +64,18 @@ const BannerImageField = ({
   headerBg,
   onAddImage,
   onRemoveImage,
-  setFormStatus,
   onOverlayChange,
 }: Props) => {
   const { formatMessage } = useIntl();
   const [previewDevice, setPreviewDevice] = useState<TPreviewDevice>('desktop');
   const [headerLocalDisplayImage, setHeaderLocalDisplayImage] =
     useState<TLocalHeaderImage>(null);
   const [bannerError, setBannerError] = useState<TBannerError>(null);
-  useEffect(() => {
-    // the image file sent from the API needs to be converted
-    // to a format that can be displayed. this is done locally
-    // when the image is changed but needs to be done manually
-    // to process the initial API response
-    const convertHeaderToUploadFile = async (fileInfo) => {
-      if (fileInfo) {
-        const tenantHeaderBg = await convertUrlToUploadFile(fileInfo);
-        const headerBgUploadFile = !isNilOrError(tenantHeaderBg)
-          ? [tenantHeaderBg]
-          : [];
-        setHeaderLocalDisplayImage(headerBgUploadFile);
-        setBannerError(null);
-      }
-    };
 
+  useEffect(() => {
+    // if (typeof headerBg === 'string') return;
     const headerFileInfo = headerBg?.large;
+
     convertHeaderToUploadFile(headerFileInfo);
   }, [headerBg]);
 
```

Comment:
There is still a problem here. headerBg returns something different when we have an unsaved vs. saved image (base64 string vs. object of type ImageSizes) but the types for header_bg in homepageSettings and customPages are not making this obvious (because they are wrong/incomplete). I'll address it in the FE CDM.